addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -4036,9 +4036,14 @@ options.registry.GridImage = options.Class.extend({
      * @override
      */
     _computeVisibility() {
+        // Special conditions for the hover effects.
+        const hasSquareShape = this.$target[0].dataset.shape === "web_editor/geometric/geo_square";
+        const effectAllowsOption = !["outline", "image_mirror_blur"].includes(this.$target[0].dataset.hoverEffect);
```

Comment:
I think the dolly zoom should also be included. With the example of the banner snippet, it doesn't work great on the main image (because of the image's shadow) nor on the other ones (shadow + rounded corners cut halfway). 
Generally speaking I guess not having the whole image defeats the purpose of the dolly zoom.

Reply:
Yes I was hesitating about it too, the effect is not as obvious as when it is "contain" :thinking: 
But when discussing with BVR, we thought that only "outline" and "mirror blur" were catastrophic and therefore should never be allowed to be "cover" (as they have borders that may never be shown) 
Here at worst, the user has the choice to change to "contain" if he finds it weird :thinking: 
I will ask if it is okay :slightly_smiling_face: 

> With the example of the banner snippet

I don't think we should take this snippet as the reference for the behavior, since it is already styled a lot, it doesn't represent the base case. However, indeed, the shadow is strange when it is contain :open_mouth: I will create a task, as it is a general issue :+1: 

> rounded corners cut halfway

Note that it is also the case in normal mode: 
- drop Text-image
- add the "Dolly Zoom" hover effect on the image and set round corners
- Hover on it
=> the zoomed image has sharp corners